server/request_pusher.go

```diff
@@ -0,0 +1,81 @@
+package server
+
+import (
+	"fmt"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/flashbots/rpc-endpoint/database"
+	"time"
+)
+
+type RequestPusher struct {
+	db         database.Store
+	EntryChan  chan database.Entry
+	BatchChan  chan []database.Entry
+	maxItem    int           // maximum item to be pushed as batch
+	maxTimeOut time.Duration // timeout in which entry to be added to batch queue
+}
+
+func NewRequestPusher(db database.Store, maxItem int, maxTimeOut time.Duration) *RequestPusher {
+	return &RequestPusher{
+		db:         db,
+		EntryChan:  make(chan database.Entry, 100),
+		BatchChan:  make(chan []database.Entry),
+		maxItem:    maxItem,
+		maxTimeOut: maxTimeOut,
+	}
+}
+
+func (r *RequestPusher) Run() {
+	go r.saveEntries()
+	go r.addEntryToEntryQueue()
+}
+
+func (r *RequestPusher) addEntryToEntryQueue() {
```

Comment:
This function is a bit of a mess :)

Usually it's best to avoid goto, except you need it. creating a new ticker every loop round is not needed (maybe a timer, but best to wait until the job is done before creating the next one).

also `keepGoing` is unnecessary, can just be a `for` loop and you break or return.